### PR TITLE
Add new package: libcgroup

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

File: var/spack/repos/builtin/packages/libcgroup/package.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    depends_on('m4',       type='build')
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('bison',    type='build')
+    depends_on('flex',     type='build')
+    depends_on('linux-pam')
+
+    def setup_build_environment(self, env):
+        env.prepend_path('CPATH', self.spec['linux-pam'].prefix.include)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libcgroup/package.py]: Why is this needed? I thought `link` typoe dependencies were injecting this automatically into the compiler wrappers.

**Review by alalazo — CHANGES_REQUESTED** [review]
A question on `setup_build_environment`

File: var/spack/repos/builtin/packages/libcgroup/package.py (hunk 1 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/libcgroup/package.py]: Fixed it in https://github.com/spack/spack/pull/16229/commits/8be628c824f00fae518a8bb513884332d20d0303

**Review by alalazo — APPROVED** [review]
